### PR TITLE
Adds feature to DSL - experiment definition

### PR DESCRIPTION
**Pull request by jeppe-style**
This adds API calls for the BenchFlow Experiment definition.

- `experimentFromTestYaml(testDefinitionYaml: String): Try[BenchFlowExperiment]`: get an Experiment object from a Test definition YAML string
- `experimentFromExperimentYaml(experimentDefinitionYaml: String): Try[BenchFlowExperiment]`: get an Experiment object from an Experiment definition YAML string
- `experimentToYamlString(benchFlowExperiment: BenchFlowExperiment): String`: get a YAML string from an Experiment object

### CODE REVIEW

File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/ExperimentOnlyTerminationCriteriaYamlProtocol.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import cloud.benchflow.dsl.definition.errorhandling.YamlErrorHandler.{ deserializationHandler, unsupportedReadOperation, unsupportedWriteOperation }
+import net.jcazevedo.moultingyaml.{ DefaultYamlProtocol, YamlFormat, YamlObject, YamlString, YamlValue, _ }
+
+import scala.util.Try
+
+/**
+ * @author Jesper Findahl (jesper.findahl@usi.ch)
+ *         created on 11.03.17.
+ */
+object ExperimentOnlyTerminationCriteriaYamlProtocol extends DefaultYamlProtocol {
```
Comment (VincenzoFerme) [inline benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/ExperimentOnlyTerminationCriteriaYamlProtocol.scala:14]: Why the same of this class is `Experiment`**Only**?

**Review by VincenzoFerme — COMMENTED** [review]
Please add some comments explaining where things happens, and how you convert a Test to an Experiment taking advantage of the library you use. 

Moreover we agreed to define and document a policy to name classes related to test and experiment: BenchFlowTest.., BenchFlowExperiment...

**Comment by VincenzoFerme** [comment]
Referring to https://github.com/benchflow/benchflow/pull/118#pullrequestreview-31256575, the refactoring steps should happen in a new pull request, related to an issue tracking what to do.

File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/ExperimentOnlyTerminationCriteriaYamlProtocol.scala (hunk 1 above)
Comment (VincenzoFerme) [reply benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/terminationcriteria/ExperimentOnlyTerminationCriteriaYamlProtocol.scala:14]: Solved by https://github.com/benchflow/benchflow/pull/118#issuecomment-292150312

**Review by VincenzoFerme — APPROVED** [review]

File: benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/BenchFlowExperimentYamlProtocol.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        workload <- deserializationHandler(
+          yamlObject.fields(WorkloadKey).convertTo[Map[String, Try[Workload]]].mapValues(_.get),
+          keyString(WorkloadKey))
+
+        dataCollection <- deserializationHandler(
+          yamlObject.getFields(DataCollectionKey).headOption.map(_.convertTo[Try[DataCollection]].get),
+          keyString(DataCollectionKey))
+
+      } yield BenchFlowExperiment(
```
Comment (igormoreno) [inline benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/BenchFlowExperimentYamlProtocol.scala:68]: Using named parameters can make the code more descriptive and understandable but I think in this case the name of the variable and the parameter are always the same so it doesn't add much information to have named parameters in this case.

**Pull request merged**